server/src/service/PkiService.js

```diff
@@ -298,6 +298,39 @@ exports.splitChainIntermediateCertificate = async (body) => {
   return chainInfo;
 };
 
+/**
+ *
+ */
+exports.retrieveFirstAndRemainingIntermediateChainCerts = async (intermediateChain) => {
+  const beginCertRegex = /(?=-----BEGIN)/g;
+
+  let chainInfo = [];
+  let remainingIntermediateChainInfo = '';
+  let count = ((intermediateChain && intermediateChain.match(/BEGIN/g)) || []).length;
+  // split the intermediatesChain into a list of certInfo
+  if (count > 0) {
+    let intermediateChains = intermediateChain.split(beginCertRegex);
+    for (let index = 0; index < intermediateChains.length; index++) {
```

Comment:
Why is it iterating through all the certs to just return the first one? I guess I don't understand what is it doing at line 315

Reply:
This logic was necessary to deal with the Amazon certificate comments in which case it would perform 2 iterations. Per your comment for non Amazon certificates or Amazon certificate that do not include comments...yes, there would be not need to iterate and it would only grab the first.